src/sage/combinat/output.py

```diff
@@ -507,3 +609,167 @@ def get_len(e):
+                return 0
+            return len(e)
+
+    # Length of max string (ensure it's odd)
+    str_tab = [[str(val) if val is not None else None for val in row] for row in data]
+    max_str = max([max([1] + [get_len(e) for e in row]) for row in str_tab])
+    max_str = max_str + 1 - (max_str % 2)
+
+    if compact:
+        diag_length = max_str
+    else:
+        diag_length = max_str + 2  # space on both sides
+
+    row_height = int((diag_length + 1)/2)
+    max_height = max([a + len(str_tab[a]) for a in range(len(str_tab))])
+    str_list = []
+    for k in range(max_height, -1, -1):
+        for i in range(row_height):
+            if k == max_height and i == 0:
+                continue
+            st = ' ' * ((max_height - k) * row_height)
+            for j in range(k + 1):
+                N_box = box_exists(str_tab, k-j+1, j)
+                S_box = box_exists(str_tab, k-j, j-1)
+                SE_box = box_exists(str_tab, k-j-1, j)
+                E_box = box_exists(str_tab, k-j, j)
+                W_box = box_exists(str_tab, k-j+1, j-1)
+                if i == 0:
+                    if (N_box and S_box) or (W_box and E_box):
+                        st += x
+                    elif (E_box and S_box) or (W_box and N_box):
+                        st += urdl
+                    elif (E_box and N_box) or (W_box and S_box):
+                        st += uldr
+                    elif E_box:
+                        st += uldr
+                    elif W_box:
+                        st += urdl
+                    else:
+                        st += ' '
+                    if E_box:
+                        st_num = str_tab[k-j][j]
+                        ln_left = int((len(st_num) - (len(st_num) % 2))/2)
+                        st += st_num.rjust(row_height - 1 - ln_left + len(st_num), ' ').ljust(diag_length, ' ')
+                    else:
+                        st += ' ' * diag_length
+                    if j == k and E_box:
+                        st += urdl
+                else:
+                    lstr = ' '
+                    rstr = ' '
+                    if E_box or S_box:
+                        lstr = uldr
+                    if E_box or SE_box:
+                        rstr = urdl
+                    st += ' ' * i
+                    st += lstr
+                    st += ' ' * (2 * (row_height - i) - 1)
+                    st += rstr
+                    st += ' ' * (i-1)
+            str_list.append(st)
+
+    import re
+    mm = min([len(re.search('^ +', l)[0]) for l in str_list]) - 1
+    str_list = [l[mm:].rstrip() for l in str_list]
+    while str_list[-1] == '':
+        str_list.pop()
+    return "\n".join(str_list)
+
+
+def box_exists(tab, i, j):
+    """
+    For Russian tableaux, it checks if a certain box exists or not.
+    """
+    if j < 0 or i < 0:
+        return False
+    return (len(tab[i:]) > 0 and len(tab[i][j:]) > 0 and tab[i][j] is not None)
```

Comment:
This function needs a doctest. I think it can also be made a bit faster by
```suggestion
def box_exists(tab, i, j):
    """
    For Russian tableaux, it checks if a certain box exists or not.
    """
    if j < 0 or i < 0:
        return False
    return len(tab) >= i and len(tab[i]) >=j and tab[i][j] is not None
```
or possibly
```
return bool(tab[i:]) and bool(tab[i][j:]) and tab[i][j] is not None
```